### ISSUE TITLE
Support R2 copy object

### ISSUE THREAD

**Issue by abersheeran** [cloudflare/workerd#1349]
The S3 API of R2 supports `CopyObject`, but the R2 interface in Workers does not support Copy operation.

**Comment by mkmaker** [comment]
can we copy an object in r2 via API?

**Comment by a-robinson** [comment]
> can we copy an object in r2 via API?

Yes, see CopyObject in [the list of supported API endpoints in the S3 API compatibility docs](https://developers.cloudflare.com/r2/api/s3/api/#implemented-object-level-operations).
